### DISCUSSION TITLE
Sobre Responsividade.

### DISCUSSION THREAD

**Discussion by baltazarparra** [frontendbr/forum#86] [Perguntas e Respostas]
**Como vocês abrangem essa questão?**

Trabalho em uma agência onde os designers vem de mídia offline _(pouca ou nenhuma base em Html/Css)_ e o workflow é um pouco engessado pela dificuldade em manter bons desenvolvedores, estamos trabalhando para mudar aos poucos, portanto mobile-first ainda é um sonho distante.

Então eles desenham o layout desktop, e eu trato o responsivo, adaptando os elementos para não haver quebra em nenhuma resolução/dispositivo, abro o devtools e vou redimensionando pixel por pixel pra checar se há quebra de layout, e vou corrigindo.

Não tenho certeza se essa é a melhor forma de tratar isso, gostaria de saber como vocês lidam com esse assunto quando o mobile-first está fora de questão.

**Comment by Cleysonlb** [comment]
@baltazarparra sei bem como é isso, até hoje só trabalhei em ambiente de agência e sim, é mt engessado, dificil de mudar a cabeça do dono da agencia e dos designs da agencia! Já passei por muita dor de cabeça com layouts exorbitantes que dificultavam muito adaptar para dispositivo mobile, porem era "obrigatório" ja que eles ofereciam responsividade no produto deles....
Você pode tentar ao máximo adaptar da melhor maneira possivel só pra não ficar feio como já esta fazendo, e aos poucos em reuniões etc tentar mudar a cabeça deles, mostrando palestras, posts, exemplos sobre o assunto!
Enfim, boa sorte :dancer:

**Comment by williamcosta** [comment]
Aqui aonde trabalho não rola mobile-first pois os prazos são meio apertados e sempre o cliente quer ver o resultado e o desenvolvimento do site na tela. Ou seja fazemos sempre o Desktop primeiro.

Também temos muito problemas com os designers. Por diversas vezes peguei layout que na versão mobile o elemento que por exemplo ficava no topo da página no mobile está posicionado no meio da página e ai temos que apelar para copiar o elemento e dar display block e none. E é bem dificil convencer a galera para fazer mobile-first.

**Comment by leandroruel** [comment]
já trabalhei em agência assim também, e exatamente da mesma maneira, que você disse, os designers eram mais de offline, sempre que tentava mudar a cabeça deles para a questão de responsividade eles me entregavam um layout pior que o outro, sempre com bolas dentro de triângulos. a melhor maneira de tratar isso é: quando chegar o layout, não deixe que enviem para o cliente aprovar layout sem que antes você dê uma olhada, se ver que tem alguma coisa que será dificil responsivar, tente convencer o designer a mudar aquilo, por exemplo, quando o cara te manda um menu redondo ou triangular e você vê que não vai rolar no mobile. se mesmo assim eles se recusarem a alterar aquilo, substitua os elementos da página por outros que possam ser responsivados com media queries, por exemplo: esconde um menu triangular e mostra um navbar, OU, altera a aparência desse menu sem precisar substituir ele no mobile. apenas usando media queries, infelizmente você passará por este tipo de empecilho em todos seus projetos, não adianta tentar mudar a cabeça deles que por experiência própria já tentei em minha agência e não funciona, o negócio é ir se virando mesmo.
